src/graphics/gui.c

```diff
@@ -5,28 +5,30 @@
 
 #include "gui.h"
 #include "state.h"
-int screenWidth = 800;
-int screenHeight = 800;
-char title[] = "Minesweeper RTS";
+#include "constants.h"
+#include "start.h"
+int screen_width = INITIAL_SCREEN_WIDTH;
+int screen_height = INITIAL_SCREEN_HEIGHT;
+char title[] = GAME_TITLE;
 
 int state = STATE_START;
 
 void runGameGui() {
-    InitWindow(screenWidth, screenHeight, title);
-    
-    SetTargetFPS(60);
 
-    while (!WindowShouldClose()) {
-        BeginDrawing();
-        ClearBackground(RAYWHITE);
-        DrawText("Example Window!", screenWidth/2 - MeasureText("Example Window!", 32) / 2, screenHeight / 2 - 32 / 2, 32, DARKGRAY);
-        switch (state){
-            default:
-                // TODO: Add an error message.
-                break;
-        }
-        EndDrawing();
-    }
+    MenusBundle *menu_bundle = initMenusBundle(&screen_width,&screen_height);
+    InitWindow(screen_width, screen_height, title);
 
+    SetTargetFPS(60);
+    Vector2 mousePos= {0.0f, 0.0f};
+    Vector2 *mp =&mousePos;
+    while (!WindowShouldClose()&&state!=STATE_EXIT) {
+        handleStateClicks(menu_bundle,mp,&state);
+        handleGUIDraw(menu_bundle,mp,&state,&screen_width,&screen_height);
+    }
+    freeSettings(menu_bundle->settings);
+    freeStartGameSettings(menu_bundle->start_settings);
+    freeStatistics(menu_bundle->statistics);
+    freeStartMenu(menu_bundle->menu);
+    freeMenusBundle(menu_bundle);
     CloseWindow();
```

Comment:
When you try to quit, this error is logged:
```
INFO: TIMER: Target time per frame: 16.667 milliseconds
double free or corruption (out)
make: *** [Makefile:30: run] Aborted (core dumped)
```

Make sure you're not freeing something twice. When programming in C, you must be super careful with your memory. You can use a program called `valgrind` to find memory issues.

Reply:
that is weird, on my computer it does not provide that error

Reply:
I have fixed the issue